executor/insert_common.go

```diff
@@ -598,7 +599,7 @@ func (e *InsertValues) fillRow(ctx context.Context, row []types.Datum, hasValue
 
 // isAutoNull can help judge whether a datum is AutoIncrement Null quickly.
 // This used to help lazyFillAutoIncrement to find consecutive N datum backwards for batch autoID alloc.
-func (e *InsertValues) isAutoNull(ctx context.Context, d types.Datum, col *table.Column) bool {
+func (e *InsertValues) isAutoNull(ctx context.Context, d *types.Datum, col *table.Column) bool {
```

Comment:
IMHO we don't need to change it to pass-by-reference in case that it may be modified carelessly in the future?